unit/testing-utils/load_java_class.cpp

```diff
@@ -62,7 +62,9 @@ symbol_tablet load_java_class(
   // Construct a lazy_goto_modelt
   null_message_handlert message_handler;
   lazy_goto_modelt lazy_goto_model(
-    [] (goto_functionst::goto_functiont &function, symbol_tablet &symbol_table)
+    [] (goto_functionst::goto_functiont &function,
+        symbol_tablet &symbol_table,
+        goto_program_location_numberingt &location_numbering)
```

Comment:
It feels quite odd nonetheless that we introduced `goto_modelt` as a container for goto_functions and symbol_table to simplify our interfaces and now here we are creating interfaces that require even more parameters. What about wrapping a goto_function and its goto_model into a `goto_function_modelt`? Then you can call renumber() on the goto_function_model without exposing the underlying goto_model.